src/Okta.Idx.Sdk/SocialLoginSettings.cs

```diff
@@ -0,0 +1,30 @@
+﻿// <copyright file="IdxState.cs" company="Okta, Inc">
+// Copyright (c) 2020 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using Newtonsoft.Json;
+using Okta.Idx.Sdk.Configuration;
+
+namespace Okta.Idx.Sdk
+{
+    /// <summary>
+    /// A descriptor for the current state of an Idx social login interaction.
+    /// </summary>
+    public class SocialLoginSettings
+    {
+        public string Key
+        {
+            get => Context.State;
+        }
+
+        public IIdxContext Context { get; set; }
+
+        public IdpOption[] IdpOptions { get; set; }
+
+        [JsonIgnore]
+        public IdxClient Client { get; set; }
```

Comment:
Since the client is returning this response, why do we need the client here? I might be missing something.

Reply:
I'll review this more closely, but the thought was that if we have an instance of this class then we'd be able to use the client that instantiated it directly from the client property of the instance itself.  This may have been useful in testing; I'll review more closely to see if this no longer useful.